truss-transfer/src/lib.rs

```diff
@@ -774,6 +803,25 @@ async fn handle_write_b10cache(download_path: &Path, cache_path: &Path) -> Resul
         download_path
     );
     Ok(())
+}// Get the desired download speed from the environment variable or use a heuristic.
+// Heuristcally, if the number of CPU cores is 64 or fewer, use a lower speed.
+// If the environment variable is not set, use a random number between 25 and 300 MB/s.
+fn get_desired_speed() -> f64 {
+    if let Ok(speed) = env::var(TRUSS_TRANSFER_B10FS_DOWNLOAD_SPEED_MBPS_ENV_VAR) {
+        if let Ok(speed) = speed.parse::<f64>() {
+            return speed;
+        }
+    }
+
+    // if we have 64 or fewer cpu cores, use a lower speed
+    let used_speed = if num_cpus::get() <= 64 {
+        TRUSS_TRANSFER_B10FS_DOWNLOAD_SPEED_MBPS_FEW_CORES
+    } else {
+        TRUSS_TRANSFER_B10FS_DOWNLOAD_SPEED_MBPS
+    };
+
+    // fallback to a random number between 25 and 300 MB/s
+    25.0 + rand::random::<f64>() * (used_speed - 25.0)
```

Comment:
nit: I think `speed_threshold` may be a more clear name
```suggestion
    let speed_threshold = if num_cpus::get() <= 64 {
        TRUSS_TRANSFER_B10FS_DOWNLOAD_SPEED_MBPS_FEW_CORES
    } else {
        TRUSS_TRANSFER_B10FS_DOWNLOAD_SPEED_MBPS
    };

    // fallback to a random number between 25 MB/s and speed_threshold
    25.0 + rand::random::<f64>() * (speed_threshold - 25.0)
```